analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -69,11 +69,20 @@ public Verifier(VerifierBuilder builder)
 
         public void Verify()    // This should never has any arguments
         {
-            using var scope = new EnvironmentVariableScope { EnableConcurrentAnalysis = true };         // ToDo: Implement properly
-            var paths = builder.Paths.Select(x => Path.GetFullPath(Path.Combine("TestCases", x)));
-            var pathsWithConcurrencyTests = paths.Count() == 1 ? CreateConcurrencyTest(paths) : paths;  // ToDo: Redesign when implementing concurrency
-            var solution = SolutionBuilder.CreateSolutionFromPaths(pathsWithConcurrencyTests, builder.OutputKind, builder.References);
-            foreach (var compilation in solution.Compile(builder.ParseOptions.ToArray()))
+            const string TestCases = "TestCases";
+            using var scope = new EnvironmentVariableScope { EnableConcurrentAnalysis = builder.ConcurrentAnalysis};
+            var basePath = Path.GetFullPath(builder.BasePath == null ? TestCases : Path.Combine(TestCases, builder.BasePath));
+            var paths = builder.Paths.Select(x => Path.Combine(basePath, x));
+            var pathsWithConcurrencyTests = paths.Count() == 1 && builder.ConcurrentAnalysis ? CreateConcurrencyTest(paths) : paths;  // ToDo: Redesign when implementing concurrency
+            var project = SolutionBuilder.Create()
+                .AddProject(language, true, builder.OutputKind)
+                .AddDocuments(pathsWithConcurrencyTests)
+                .AddReferences(builder.References);
+            foreach (var snippet in builder.Snippets)
+            {
+                project = project.AddSnippet(snippet);
```

Comment:
Shouldn't we consider an API which can handle multiple snippets directly? We would be able to make at least this method a bit cleaner.